web/index.js

```diff
@@ -242,11 +241,6 @@ function createBotRoles(data, rolesFromSettings) {
       'participantsBotRoles', serialize(rolesFromSettings))
       .then((res) => rolesBotDataId = res.body.id);
   }
-
-  roles.forEach((role) => {
-    currentRole[role.label] = data.body
```

Comment:
since currentRole has instance scope I'd make sure who is using it, this _for_  is updating all the current roles.

Reply:
👍  yeah, that's the colateral effect I'm trying to figure it out. I'll spend bit more of time on it today.

Reply:
Found a potential problem if the for each is removed. If the ParticipantsRoles botData is re-created and not loaded on the UI, hence we need to update the currentRole. Not sure on which case the ParticipantRoles could be removed. But that gave me a better understanding on what is needed.